ntoskrnl/se/srm.c

```diff
@@ -376,17 +381,96 @@ SepRmCreateLogonSession(
     return Status;
 }
 
+/**
+ * @brief
+ * Deletes a logon session from the logon sessions database.
+ * 
+ * @param[in] LogonLuid
+ * A logon ID represented as a LUID. This LUID is used to point
+ * the exact logon session saved within the database.
+ * 
+ * @return
+ * STATUS_SUCCESS is returned if the logon session has been deleted successfully.
+ * STATUS_NO_SUCH_LOGON_SESSION is returned if the logon session with the submitted
+ * LUID doesn't exist. STATUS_BAD_LOGON_SESSION_STATE is returned if the logon session
+ * is still in use and we're not allowed to delete it. Otherwise a NTSTATUS failure code
+ * is returned.
+ */
 static
 NTSTATUS
 SepRmDeleteLogonSession(
-    PLUID LogonLuid)
+    _In_ PLUID LogonLuid)
 {
+    PSEP_LOGON_SESSION_REFERENCES SessionToDelete;
+    NTSTATUS Status;
+    PAGED_CODE();
+
     DPRINT("SepRmDeleteLogonSession(%08lx:%08lx)\n",
            LogonLuid->HighPart, LogonLuid->LowPart);
 
-    UNIMPLEMENTED;
-    NT_ASSERT(FALSE);
-    return STATUS_NOT_IMPLEMENTED;
+    /* Acquire the database lock */
+    KeAcquireGuardedMutex(&SepRmDbLock);
+
+    /* Loop over the existing logon sessions */
+    for (SessionToDelete = SepLogonSessions;
+         SessionToDelete != NULL;
+         SessionToDelete = SessionToDelete->Next)
+    {
+        /*
+         * Does the actual logon session exist in the
+         * saved logon sessions database with the LUID
+         * provided?
+         */
+        if (!RtlEqualLuid(&SessionToDelete->LogonId, LogonLuid))
```

Comment:
It might cause troubles when multiple logon session will be set, and if we iterate on all of this, one can be different.
Then we skip other in list, and targeted one might be missed

Reply:
Yeah, I realized quitting the loop aggressively which could leave behind potential logon sessions with the expected LUID can definitely lead to problems. I'll correct it asap.